### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by mark-likhtar**
- contains SignIn/SignUp components 
- SignIn contains login and password fields, component request on server endpoint to find user with correct information, if all ok, redirect to main page
- SignUp component contains name, login, password and repeatPassword fields and request on server to write user on database, check all errors

### CODE REVIEW

File: src/components/Auth/Auth.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            <LogIn action={this.toggleForm} snackbar={this.openSnackbar} />
+          ) : (
+            <SignUp action={this.toggleForm} snackbar={this.openSnackbar}/>
+          )}
+          <Snackbar
+            anchorOrigin={{ vertical, horizontal }}
+            open={openSnack}
+            onClose={this.handleClose}
+            ContentProps={{
+              'aria-describedby': 'message-id'
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Auth/Auth.js]: What is this for?

File: src/components/Auth/style.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  align-items: center;
+  height: 90vh;
+}
+
+.auth__form {
+  background-color: #fff;
+  border-radius: 4px;
+  padding: 2%;
+  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2),
+    0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 2px 1px -1px rgba(0, 0, 0, 0.12);
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Auth/style.scss]: formatting

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -6,9 +6,18 @@ export default class Header extends Component {
   state = {
     isAdmin: false,
     logIn: false,
-    userName: 'UserName'
+    userName: ''
   };
 
+  componentWillMount = () => {
+    const userName = localStorage.getItem('userName');
+    userName ? this.setState({logIn: true, userName}) : this.setState({logIn: false})
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Header/Header.js]: Why are you using user name to determine if user is logged in?

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -6,9 +6,18 @@ export default class Header extends Component {
...
+  }
+
+  componentWillReceiveProps = () => {
+    const userName = localStorage.getItem('userName');
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Header/Header.js]: why is the same code here as in componentWillMount() ?

File: src/components/SignUp/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    repeatPasswod: ''
+  };
+
+  handleChange = name => event => {
+    this.setState({ [name]: event.target.value });
+  };
+
+  loginSubmit = () => {
+    const { name, login, password, repeatPasswod } = this.state;
+    !name || !login || !password || !repeatPasswod
```
Comment (andrei-kruhliakou-itechart) [inline src/components/SignUp/SignUp.js]: Simplify structure of this statement. It's very hard to follow all those ? : . It's better to use separate if statements here (Note: I'm not saying not to use ternary operator at all but you should think about code readability first)

File: src/components/LogIn/LogIn.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    login && password
+      ? axios
+          .post('http://localhost:3000/api/login', {
+            login,
+            password
+          })
+          .then(res => {
+            localStorage.setItem('token', res.data.token);
+            localStorage.setItem('userName', res.data.userName);
+            localStorage.setItem('userLogin', res.data.userLogin);
```
Comment (andrei-kruhliakou-itechart) [inline src/components/LogIn/LogIn.js]: Why do you need to store login for?

File: src/components/SessionCard/SessionCard.js [hunk 7]
```diff
@@ -58,30 +41,46 @@ export default class SessionCard extends Component {
...
       })
       .catch(() => {
         this.setState({
-          errorMessage: 'Не удалось заказать билет'
+          snackMessage: 'Не удалось заказать', 
+          open: false,
+          openSnack: true
         });
+        
+        setTimeout(() =>
```
Comment (andrei-kruhliakou-itechart) [inline src/components/SessionCard/SessionCard.js]: indentation

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.js [hunk 8]
```diff
@@ -1,25 +1,49 @@
...
+import MenuItem from '@material-ui/core/MenuItem';
 import style from './style.scss';
 
 export default class Header extends Component {
   state = {
     isAdmin: false,
     logIn: false,
-    userName: ''
+    userName: '',
+    anchorEl: null
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Header/Header.js]: What this property for? Name it properly.
Also working with DOM elements directly doesn't feel natural in React. Try to think of another approach.

File: src/components/Header/Header.js [hunk 9]
```diff
@@ -1,25 +1,49 @@
...
     const userName = localStorage.getItem('userName');
-    userName ? this.setState({logIn: true, userName}) : this.setState({logIn: false})
-  }
+    const token = localStorage.getItem('token');
+    token
+      ? this.setState({ logIn: true, userName })
+      : this.setState({ logIn: false });
+  };
 
   componentWillReceiveProps = () => {
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Header/Header.js]: componentWillMount and componentWillReceiveProps are both deprecated
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

File: src/components/Header/Header.js [hunk 10]
```diff
@@ -30,9 +54,51 @@ export default class Header extends Component {
         <div className="header__user">
           {logIn ? (
             isAdmin ? (
-              <Link to="/admin">{userName}</Link>
+              <>
+                <Button
+                  aria-owns={anchorEl ? 'simple-menu' : undefined}
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Header/Header.js]: please STOP copypasting code without understanding what it is doing!

File: src/components/UserPage/UserPage.js [hunk 11]
```diff
@@ -1,7 +1,24 @@
...
+  componentWillMount = () => {
+    const token = localStorage.getItem('token');
+    axios
+      .get('http://localhost:3000/api/auth', {
+        headers: {
+          Authorization: 'JWT ' + token,
+          'Content-Type': 'application/x-www-form-urlencoded'
+        }
+      })
+      .catch(() => {
```
Comment (andrei-kruhliakou-itechart) [inline src/components/UserPage/UserPage.js]: You shouldn't catch them all, login related data should be changed only on 401

File: src/components/Header/Header.js [hunk 12]
```diff
@@ -1,25 +1,49 @@
...
+    this.setState({ anchorEl: event.currentTarget });
+  };
+
+  handleClose = () => {
+    this.setState({ anchorEl: null });
+  };
+
+  logOut = () => {
+    this.setState({ anchorEl: null });
+    localStorage.removeItem('token');
```
Comment (andrei-kruhliakou-itechart) [inline src/components/Header/Header.js]: Move all the code related to storing and retrieving auth data from local storage to a separate service class

File: src/components/UserPage/UserPage.js [hunk 13]
```diff
@@ -1,7 +1,24 @@
 import React, { Component } from 'react';
+import axios from 'axios';
+import history from '../../history';
 import style from './style.scss';
 
 export default class UserPage extends Component {
+  componentWillMount = () => {
+    const token = localStorage.getItem('token');
+    axios
+      .get('http://localhost:3000/api/auth', {
```
Comment (andrei-kruhliakou-itechart) [inline src/components/UserPage/UserPage.js]: What is the purpose of this call?

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: src/Service/UserService.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+UserService.logout = () => {
+  localStorage.removeItem('token');
+  localStorage.removeItem('userName');
+};
+
+UserService.getUserName = () => {
+  return localStorage.getItem('userName');
+};
+
+UserService.isLoggedIn = () => {
```
Comment (andrei-kruhliakou-itechart) [inline src/Service/UserService.js:16]: method which name started with 'is' should return boolean value
if you need method to return a token you should create separate method and name it accordingly

File: src/components/SessionCard/SessionCard.js [hunk 15]
```diff
@@ -27,7 +28,8 @@ export default class SessionCard extends Component {
   };
 
   handleClickOpen = () => {
-    this.setState({ open: true });
+    const token = localStorage.getItem('token');
```
Comment (andrei-kruhliakou-itechart) [inline src/components/SessionCard/SessionCard.js]: use UserService method here

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: src/components/SessionCard/SessionCard.js (hunk 15 above)
Comment (mark-likhtar) [reply src/components/SessionCard/SessionCard.js]: done

**Review by andrei-kruhliakou-itechart — APPROVED** [review]
